source/expectations/data_values/ut_compound_data_helper.pkb

```diff
@@ -541,5 +541,169 @@ create or replace package body ut_compound_data_helper is
     return l_no_missing_keys;
   end;
    
+  procedure update_row_and_pk_hash(a_self_data_id in raw, a_other_data_id in raw, a_exclude_xpath varchar2, 
+                                   a_include_xpath varchar2, a_join_by_xpath varchar2) is
+    l_ut_owner        varchar2(250) := ut_utils.ut_owner;
+    l_column_filter   varchar2(32767);
+    l_pk_hash_sql     varchar2(32767);
+    
+    function get_column_pk_hash(a_join_by_xpath varchar2) return varchar2 is
+      l_column varchar2(32767);
+    begin
+      /* due to possibility of key being to columns we cannot use xmlextractvalue
+         usage of xmlagg is possible however it greatly complicates code and performance is impacted.
+         xpath to be looked at or regex
+      */
+      if a_join_by_xpath is not null then
+        l_column :=  l_ut_owner ||'.ut_compound_data_helper.get_hash(extract(ucd.item_data,:join_by_xpath).GetClobVal()) pk_hash';
+      else
+        l_column := ':join_by_xpath pk_hash';
+      end if;
+      return l_column;
+    end;  
+  
+  begin
+    l_column_filter := ut_compound_data_helper.get_columns_filter(a_exclude_xpath, a_include_xpath);
+    l_pk_hash_sql := get_column_pk_hash(a_join_by_xpath);
+         
+    execute immediate 'merge into ' || l_ut_owner || '.ut_compound_data_tmp tgt
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: An illegal character with code point 10 was found in this literal. ([plsql:LiteralsNonPrintableCharactersCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3ALiteralsNonPrintableCharactersCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=774&issues=AWbTzUDbO08i52vd_XcX&open=AWbTzUDbO08i52vd_XcX)

[ISSUE_KEY]: AWbTzUDbO08i52vd_XcX